### DISCUSSION TITLE
微信小程序Skyline渲染引擎讨论

### DISCUSSION THREAD

**Discussion by spiritree** [NervJS/taro#12238] [生态] [answered]
[Skyline 渲染引擎 Beta](https://developers.weixin.qq.com/miniprogram/dev/framework/runtime/skyline/introduction.html)
目前看来在index.config.ts加上，就可以编译到对应json文件中了
```typescript
// skyline 渲染
{
    "renderer": "skyline"
}
// webview 渲染
{
    "renderer": "webview"
}
```
不确定Taro是否需要提供额外支持，欢迎讨论

**Comment by airqj** [comment]
看着很不错   
希望能支持

**Comment by Chen-jj** [comment, marked as answer]
粗略把文档过了一遍，除去 Skyline 与 Webview 渲染自身的差异不谈，Taro 需要兼容的地方：

- 部分组件的属性
- 事件回调支持 worklet
- 兼容手势组件

更多细节需要体验过才能知道了

**Reply by airqj** [reply]
> 粗略把文档过了一遍，除去 Skyline 与 Webview 渲染自身的差异不谈，Taro 需要兼容的地方：
> 
> * 部分组件的属性
> * 事件回调支持 worklet
> * 兼容手势组件
> 
> 更多细节需要体验过才能知道了

3.5的发行日志也说下一步的工作包括了小程序的性能优化   
不知道是否会投入精力去适配skyline渲染引擎？

**Comment by sfdux** [comment]
希望Taro能支持

**Comment by youmengme** [comment]
希望能支持

**Comment by tourze** [comment]
希望能支持

**Comment by vo2-hlc-ReggieCAI** [comment]
有在开发日程里吗？这个功能也推出很久了，效果确实不错，如果taro不支持的话只能拥抱原生了

**Comment by vickyleu** [comment]
就是flutter 的skia引擎吗,切换后页面全都乱了,点击事件都完犊子

**Comment by StarCoderLn** [comment]
希望 taro 也能支持 skyline

**Comment by hdyQAQ** [comment]
希望Taro能支持

**Comment by TheKonka** [comment]
看了一下手势部分，主要是有些event callback 没有携带 `currentTarget` ，这样直接用 `eh`来代理无法获取到对应绑定callback的节点

**Comment by tutuxxx** [comment]
手势和worklet已支持，详见：https://docs.taro.zone/docs/skyline

**Reply by EvanJin** [reply]
虽然我看完了，但是在 ts 的场景下，behavior 怎么写？
